woocommerce/woocommerce
# Introduce a new block style for the Product Details block and set it as default

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Closes #41775

#### Summary

This pull request aims to enhance the design options for merchants by adding a new block style to the Product Details block. Since this block renders classic Woo content and won't be divided into smaller blocks, a block style seems like a good fit to offer additional design options.

![KSKxdq.png](https://github.com/woocommerce/woocommerce/assets/1105590/156089bc-96bf-44a3-99f1-7b9ea09092c4)

#### Specification

Using a new style will help prevent interference with existing block themes that already have their own styles for tabs and provide a better default in the single product template. Some benefits:

- Existing stores won't be affected, ensuring backward compatibility.
- New stores will default to a new global-styles-friendly style, improving the overall user experience.
- CSS rules will be limited under the `is-style-[style-name]` classes, minimizing the risk of conflicts.

The block styles are:

- **Classic**: Identical to the previous style, useful for backward compatibility with themes.
- **Minimal**: A simple style with a bottom border on the selected item.

#### Screenshots

##### Classic (current)

|Desktop|Mobile|Frontend|
|---|---|---|
|![pbCkOR](https://github.com/woocommerce/woocommerce/assets/1105590/29e0cf45-2b31-4a16-9edc-7f500afa0ed7)|![DgSCUO.png](https://github.com/woocommerce/woocommerce/assets/1105590/3f9d157b-b480-4ba7-9604-5be6926e7e81)|![2h7GfN.png](https://github.com/woocommerce/woocommerce/assets/1105590/7da5e4c0-f43a-410c-800d-48c41e467137)|

##### Minimal

|Desktop|Mobile|Frontend|
|---|---|---|
|![kGbtuK.png](https://github.com/woocommerce/woocommerce/assets/1105590/d77e1dec-0603-4f93-8eb9-01e8961e20c8)|![nnq5kX.png](https://github.com/woocommerce/woocommerce/assets/1105590/451cbad4-de11-4e66-b738-60b72f0412eb)|![IX9Z4G.png](https://github.com/woocommerce/woocommerce/assets/1105590/23dde9ab-5cca-457e-a74d-c58e63395dac)|

<details>
<summary>Modern (dropped)</summary>

##### Modern

|Desktop|Mobile|Frontend|
|---|---|---|
|![P0NqNG.png](https://github.com/woocommerce/woocommerce/assets/1105590/2356f7dd-e20b-4052-ad9b-d99095ef8e97)|![nbqFCD.png](https://github.com/woocommerce/woocommerce/assets/1105590/4dc19350-1a7d-4a68-b498-c8e8b94f0643)|![7pTCcO.png](https://github.com/woocommerce/woocommerce/assets/1105590/c96687d3-27e8-460a-9887-ef88eb303b0f)|
</details>

In terms of design, this is a fairly basic style, providing a minimal and easily controllable result **through color inheritance**. If you have any suggestions, please feel free to share them. In terms of style naming, suggestions are also welcome!



### Validation

#### Testing the new block style

1. Use a site with a block theme (ideally TT4, and TT3.)
2. Edit the Single Product template and notice the tabs styling being the classic one.
3. Apply this patch and refresh.
4. Ensure that you still get the classic style.
5. Select the Product Details block and change the style to Minimal.
6. Ensure that it's working as expected (both in desktop and mobile views)

#### Testing the "Transform into blocks" button

1. From the inspector controls, choose the Template option and click on the "Revert to Classic Product Template" button.
1. Select the placeholder block and click on the "Transform into blocks" button.
2. Notice the new style applied.

#### Testing the default template

1. Open the site editor and navigate to the single product template again.
3. Ensure that the tabs are using the classic style.
4. Navigate to Appearance > Editor > Templates > Manage all templates
5. Locate the "Single Product" template, and from the three-dots menu on the right, select "Clear customizations"
6. Navigate back to the single product template in the editor and notice the new style applied.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
